src/dex/uniswap-v2/uniswap-v2.ts

```diff
@@ -537,32 +540,56 @@ export class UniswapV2
         return null;
 
       await this.batchCatchUpPairs([[from, to]], blockNumber);
-
-      const pairParam = await this.getPairOrderedParams(from, to, blockNumber);
+      const isSell = side === SwapSide.SELL;
+      const pairParam = await this.getPairOrderedParams(
+        from,
+        to,
+        blockNumber,
+        transferFees.srcDexFee,
+      );
 
       if (!pairParam) return null;
 
-      const unitAmount = getBigIntPow(
-        side == SwapSide.BUY ? to.decimals : from.decimals,
+      const unitAmount = getBigIntPow(isSell ? from.decimals : to.decimals);
+
+      const [unitWithFee, ...amountsWithFee] = applyTransferFee(
+        [unitAmount, ...amounts],
+        side,
+        transferFees.srcFee,
+        SRC_TOKEN_PARASWAP_TRANSFERS,
       );
```

Comment:
Need to account for BUY case, where buyFee and buyDexFee should be applied.

Reply:
I think I fixed it